File: 0001-two-sum/0001-two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var dict = [Int: Int]()
        for (indx, value) in nums.enumerated(){
            let diff = target-value;

            if let found = dict[diff] {
                return [found,indx]
            }
            dict[value]=indx
        }

        return []
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mp;
        int sum = 0;
        vector<int> ans;
        
        for(int i = 0; i < nums.size(); i++)
        {
            int diff = target - nums[i];
            if(mp.find(diff) != mp.end())
            {
                ans.push_back(i);
                ans.push_back(mp[diff]);
                return ans;
            }
            mp[nums[i]] = i;
        }
        return ans;
    }
};